test/e2e/set/nodeconfig/nodeconfig_optimizations.go

```diff
@@ -400,7 +421,20 @@ var _ = g.Describe("NodeConfig Optimizations", framework.Serial, func() {
 		sc, err = controllerhelpers.WaitForScyllaClusterState(ctx4, f.ScyllaClient().ScyllaV1().ScyllaClusters(sc.Namespace), sc.Name, controllerhelpers.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		scyllaclusterverification.Verify(ctx, f.KubeClient(), f.ScyllaClient(), sc)
+		scyllaclusterverification.VerifyWithOptions(
+			ctx,
+			f.KubeClient(),
+			f.ScyllaClient(),
+			sc,
+			scyllaclusterverification.VerifyOptions{
+				VerifyStatefulSetOptions: scyllaclusterverification.VerifyStatefulSetOptions{
+					PodRestartCountAssertion: func(a o.Assertion, containerName, podName string) {
+						// We expect restart(s) from the startup probe, usually 1.
+						a.To(o.BeNumerically("<=", 2), fmt.Sprintf("container %q in pod %q should not be restarted by the startup probe more than twice", containerName, podName))
```

Comment:
Not sure I understand this right - isn't this testing a side effect? If you lowered the period of the startup probe, or the timeout (which according to [this TODO](https://github.com/scylladb/scylla-operator/blob/d4da9dfe44e55d7dd674a3d0394eafb5f309af2f/pkg/controller/scylladbdatacenter/resource.go#L784) should have already had been done), won't it introduce flakiness?

Reply:
Nope, my understanding is that this would straight fail and you'd need to adjust the value. Similarly the 10m30s sleep should be adjusted as well.

Reply:
ok, let's give this a try